pkg/imgpkg/bundle/bundle.go

```diff
@@ -231,3 +262,34 @@ func (o *Bundle) checkedImage() (regv1.Image, error) {
 	}
 	return img, err
 }
+
+type cachedLocationFetcher struct {
```

Comment:
I don't understand the concern or reason to change this part of the code. 

But It sounds like you are asking is there a reason I didn't instead just keep a variable reference to the `ImageLocationsConfig` and *probably* `error`? and pass those variables around to the things that need it? and call it a CachedLoctions object?

I think that would work too.

The rationale I had originally with `cachedLocationFetcher` was that there is already an API defined to `fetch` an `ImageLocationsConfig` and an `error` for a specific bundle. So decorate that with a simple caching layer since it is used for 2 reasons. Hence avoiding an unnecessary http call since the Location OCI shouldn't change between those 2 calls.